src/components/admin/Judging/JudgingSchedule.tsx

```diff
@@ -0,0 +1,152 @@
+"use client";
+
+import { generateClient } from "aws-amplify/api";
+
+import { type Schema } from "@/amplify/data/resource";
+import JudgingTimeline from "@/components/admin/Judging/JudgingTimeline";
+import RoomAssigner from "@/components/admin/Judging/RoomAssigner";
+import { useMutation, useQuery } from "@tanstack/react-query";
+
+const client = generateClient<Schema>();
+
+export default function JudgingSchedule() {
+  const { mutate } = useMutation({
+    mutationFn: async ({
+      judgingSessionsPerTeam,
+      numOfJudgingRooms,
+      startDateAndTime,
+      presentationDuration,
+    }: {
+      judgingSessionsPerTeam: number;
+      numOfJudgingRooms: number;
+      startDateAndTime: string;
+      presentationDuration: number;
+    }) => {
+      try {
+        const response = await client.mutations.ScheduleTeamsAndJudges({
+          judgingSessionsPerTeam,
+          numOfJudgingRooms,
+          startDateAndTime,
+          presentationDuration,
+        });
+
+        if (response.errors) {
+          throw new Error(response.errors[0].message);
+        }
+
+        if (response.data && response.data.body) {
+          const responseBody =
+            typeof response.data.body === "string"
+              ? response.data.body
+              : JSON.stringify(response.data.body);
+          return JSON.parse(responseBody);
+        } else {
+          console.error("Response data or body is null or undefined");
+        }
+      } catch (error) {
+        console.error("Error fetching the judging schedule: ", error);
+        throw error;
+      }
+    },
+  });
+
+  const { data: roomData } = useQuery({
+    queryKey: ["roomData"],
+    queryFn: async () => {
+      const { data, errors } = await client.models.Room.list();
+      if (errors) {
+        throw errors;
+      }
+      return data;
+    },
+  });
+
+  const { data: teamRoomData } = useQuery({
+    queryKey: ["teamRoomData"],
+    queryFn: async () => {
+      const { data, errors } = await client.models.TeamRoom.list();
+      if (errors) {
+        throw errors;
+      }
+      return data;
+    },
+  });
+
+  const { data: judgeData } = useQuery({
+    queryKey: ["judgeData"],
+    queryFn: async () => {
+      const { data, errors } = await client.models.User.list({
+        filter: {
+          role: {
+            eq: "Judge",
+          },
+        },
+        selectionSet: ["id", "JUDGE_roomId", "firstName"],
+      });
+      if (errors) {
+        throw errors;
+      }
+      return data;
+    },
+  });
+
+  const { data: teamData } = useQuery({
+    queryKey: ["teamData"],
+
+    queryFn: async () => {
+      const { data, errors } = await client.models.Team.list({
+        selectionSet: ["id", "name"],
+      });
+      if (errors) {
+        throw errors;
+      }
+      return data;
+    },
+  });
```

Comment:
Is there a way to group these queries???? Unless we want to keep them separate because of their queryKeys

Also should we invalidate queries when we submit how many rooms we want because it doesnt update automatically which I thout it would have

Reply:
Yes separate is good because of that. And yes to invalidating.